src/test/java/wedlog/address/logic/parser/VendorFilterCommandParserTest.java

```diff
@@ -0,0 +1,97 @@
+package wedlog.address.logic.parser;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static wedlog.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static wedlog.address.logic.Messages.MESSAGE_NO_PREFIX_FOUND;
+import static wedlog.address.logic.commands.CommandTestUtil.EMAIL_DESC_VAL;
+import static wedlog.address.logic.commands.CommandTestUtil.PHONE_DESC_VAL;
+import static wedlog.address.logic.commands.CommandTestUtil.PREAMBLE_NON_EMPTY;
+import static wedlog.address.logic.commands.CommandTestUtil.VALID_EMAIL_VAL;
+import static wedlog.address.logic.commands.CommandTestUtil.VALID_PHONE_VAL;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static wedlog.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import wedlog.address.logic.commands.VendorFilterCommand;
+import wedlog.address.logic.parser.exceptions.ParseException;
+import wedlog.address.model.person.AddressPredicate;
+import wedlog.address.model.person.EmailPredicate;
+import wedlog.address.model.person.NamePredicate;
+import wedlog.address.model.person.PhonePredicate;
+import wedlog.address.model.person.Vendor;
+
+class VendorFilterCommandParserTest {
+    private VendorFilterCommandParser parser = new VendorFilterCommandParser();
+
+    @Test
+    public void parse_invalidValue_failure() {
+        // no predicates
+        assertParseFailure(parser, " ", String.format(MESSAGE_NO_PREFIX_FOUND,
+                VendorFilterCommand.MESSAGE_USAGE));
+
+        // non empty preamble
+        assertParseFailure(parser, PREAMBLE_NON_EMPTY,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, VendorFilterCommand.MESSAGE_USAGE));
+
+        // empty name
+        assertParseFailure(parser, " " + PREFIX_NAME, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        "Cannot filter for empty compulsory field."));
+
```

Comment:
```suggestion

        // empty rsvp
        assertParseFailure(parser, " " + PREFIX_RSVP, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                        "Cannot filter for empty compulsory field."));
                        
```

Reply:
Vendors do not have RSVP fields, so it does not make sense to test for that here.